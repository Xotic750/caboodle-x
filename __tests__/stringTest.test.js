import {stringTest} from '../dist/caboodle-x';

describe('stringTest', () => {
  it('is a function', () => {
    expect(typeof stringTest).toBe('function');
  });

  it('should throw when target is null or undefined', () => {
    expect(() => {
      stringTest();
    }).toThrowErrorMatchingSnapshot();

    expect(() => {
      stringTest(undefined);
    }).toThrowErrorMatchingSnapshot();

    expect(() => {
      stringTest(null);
    }).toThrowErrorMatchingSnapshot();
  });

  it('should throw when search element is null or undefined', () => {
    expect(() => {
      stringTest('abc');
    }).toThrowErrorMatchingSnapshot();

    expect(() => {
      stringTest('abc', undefined);
    }).toThrowErrorMatchingSnapshot();

    expect(() => {
      stringTest('abc', null);
    }).toThrowErrorMatchingSnapshot();
  });

  it('should be true for correct string searches', () => {
    expect(stringTest('test', 'es')).toBe(true);
    expect(stringTest('abc', 'a')).toBe(true);
    expect(stringTest('abc', 'b')).toBe(true);
    expect(stringTest('abc', 'abc')).toBe(true);
    expect(stringTest('abc', 'bc')).toBe(true);
  });

  it('should be false for incorrect string searches', () => {
    expect(stringTest('test', '1290')).toBe(false);
    expect(stringTest('abc', 'd')).toBe(false);
    expect(stringTest('abc', 'abcd')).toBe(false);
    expect(stringTest('abc', 'ac')).toBe(false);
  });

  it('should be true for correct regex searches', () => {
    expect(stringTest('test', /es/)).toBe(true);
    expect(stringTest('abc', /a/)).toBe(true);
    expect(stringTest('abc', /b/)).toBe(true);
    expect(stringTest('abc', /a.c/)).toBe(true);
    expect(stringTest('abc', /bc/)).toBe(true);
  });

  it('should be false for incorrect regex searches', () => {
    expect(stringTest('test', /1290/)).toBe(false);
    expect(stringTest('abc', /d/)).toBe(false);
    expect(stringTest('abc', /ab.d/)).toBe(false);
    expect(stringTest('abc', /ac/)).toBe(false);
  });
});
