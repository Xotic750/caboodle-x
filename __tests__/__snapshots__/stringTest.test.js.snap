// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stringTest should throw when search element is null or undefined 1`] = `"Cannot call method on null or undefined"`;

exports[`stringTest should throw when search element is null or undefined 2`] = `"Cannot call method on null or undefined"`;

exports[`stringTest should throw when search element is null or undefined 3`] = `"Cannot call method on null or undefined"`;

exports[`stringTest should throw when target is null or undefined 1`] = `"Cannot call method on null or undefined"`;

exports[`stringTest should throw when target is null or undefined 2`] = `"Cannot call method on null or undefined"`;

exports[`stringTest should throw when target is null or undefined 3`] = `"Cannot call method on null or undefined"`;
